tests/CMakeLists.txt

```
@@ -7,52 +7,91 @@ endforeach()
+target_link_libraries(TestDataModelDict PUBLIC TestDataModel podio::podio ROOT::Core)
+target_include_directories(TestDataModelDict PUBLIC
+  ${CMAKE_SOURCE_DIR}/include
+  ${CMAKE_CURRENT_SOURCE_DIR}/
+ )
+
+PODIO_GENERATE_DICTIONARY(TestDataModelDict ${headers} SELECTION src/selection.xml
+  OPTIONS --library ${CMAKE_SHARED_LIBRARY_PREFIX}TestDataModelDict${CMAKE_SHARED_LIBRARY_SUFFIX}
   )
-set_target_properties(TestDataModel-dictgen PROPERTIES EXCLUDE_FROM_ALL TRUE)
-target_sources(TestDataModel PRIVATE TestDataModel.cxx)
+set_target_properties(TestDataModelDict-dictgen PROPERTIES EXCLUDE_FROM_ALL TRUE)
+add_dependencies(TestDataModelDict TestDataModel)
 
-file(GLOB executables RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.cpp)
+if (ENABLE_SIO)
+  add_library(TestDataModelSioBlocks SHARED ${sio_sources} ${sio_headers})
+  target_link_libraries(TestDataModelSioBlocks PUBLIC podio::podio)
+  target_include_directories(TestDataModelSioBlocks PUBLIC
+    ${CMAKE_SOURCE_DIR}/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/
+    ${SIO_INCLUDE_DIRS}
+  )
+  # add_dependencies(TestDataModelSioBlocks TestDataModel)
+endif()
 
-foreach( sourcefile ${executables} )
+#--- small utility helper function to allow for a more terse definition of tests below
+function(CREATE_PODIO_TEST sourcefile additional_libs)
   string( REPLACE ".cpp" "" name ${sourcefile} )
   add_executable( ${name} ${sourcefile} )
-  target_link_libraries( ${name} TestDataModel podio::podioRootIO)
-endforeach( sourcefile ${executables} )
-#--- Adding tests --------------------------------------------------------------
-
-add_test(NAME write COMMAND write)
-# The following directories need to be added to the LD_LIBRARY_PATH:
-# - CMAKE_CURRENT_BINARY_DIR: contains the root dictionary and pcm files for TestDataModel
-# - CMAKE_BINARY_DIR/src: contains the root dictionary and pcm files for podio
-set_property(TEST write PROPERTY ENVIRONMENT LD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}:${CMAKE_BINARY_DIR}/src:$ENV{LD_LIBRARY_PATH})
-
-add_test(NAME read COMMAND read)
-set_property(TEST read PROPERTY ENVIRONMENT
-             LD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}:${CMAKE_BINARY_DIR}/src:$ENV{LD_LIBRARY_PATH}
-             ROOT_INCLUDE_PATH=${CMAKE_SOURCE_DIR}/tests/datamodel:${ROOT_INCLUDE_PATH})
-set_property(TEST read PROPERTY DEPENDS write)
+  add_test(NAME ${name} COMMAND ${name})
+
+  target_link_libraries(${name} TestDataModel ${additional_libs})
+  set_property(TEST ${name} PROPERTY ENVIRONMENT
+      LD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}:${CMAKE_BINARY_DIR}/src:$ENV{LD_LIBRARY_PATH}
+      ${additional_env}
+    )
+endfunction()
+
+set(root_dependent_tests write.cpp read.cpp read-multiple.cpp relation_range.cpp)
+set(root_libs TestDataModelDict podio::podioRootIO)
+set(root_env ROOT_INCLUDE_PATH=${CMAKE_SOURCE_DIR}/tests/datamodel:${ROOT_INCLUDE_PATH})
+foreach( sourcefile ${root_dependent_tests} )
+  CREATE_PODIO_TEST(${sourcefile} "${root_libs}" ${root_env})
```

```suggestion
  CREATE_PODIO_TEST(${sourcefile} "${root_libs}")
```
Given the signature of CREATE_PODIO_TEST `root_env` isn't needed, or `additional_env` is missing there.

(I was wondering if it should be `$ENV{ROOT_INCLUDE_PATH}` in the line above, but if it is not used it doesn't matter.)

---

Another remnant from my intermediate changes. Originally I think I had `additional_env` as argument to `CREATE_PODIO_TEST`, just for passing in the `ROOT_INCLUDE_PATH` but since that is not necessary, I will completely remove all traces of it, also for the other tests.